src/Voron/Impl/Backup/FullBackup.cs

```diff
@@ -92,17 +92,21 @@ public class StorageEnvironmentInformation
             long lastWrittenLogPage = -1;
             long lastWrittenLogFile = -1;
             LowLevelTransaction txr = null;
+            Stream dataFileStream = null;
             var backupSuccess = false;
 
             try
             {
                 long allocatedPages;
+
                 var writePersistentContext = new TransactionPersistentContext(true);
                 var readPersistentContext = new TransactionPersistentContext(true);
+
                 using (env.Journal.Applicator.TakeFlushingLock()) // prevent from running JournalApplicator.UpdateDatabaseStateAfterSync() concurrently
                 using (var txw = env.NewLowLevelTransaction(writePersistentContext, TransactionFlags.ReadWrite)) // so no new journal files will be created
                 {
                     txr = env.NewLowLevelTransaction(readPersistentContext, TransactionFlags.Read);// now have snapshot view
+                    dataFileStream = File.Open(dataPager.FileName.FullPath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
```

Comment:
This is _NOT_ safe to do

Reply:
I'm opening the FileStream under the read tx which is open through the entire snapshot process.
This means that flush operations will happen up to the current read tx.
There might be a sync operation that might take place while I open the file but the current implementation suffers from the same issue.
It's "ok" to do that because we also copy the journal files and they will be applied after the database is restored.

Reply:
what about doing `msync()` after read tx, but before the `File.Open`? (not sure about the windows eqv. though)

Reply:
probably won't work, there is no guarantees that the view between mmap and `ReadFile` will be coherent in Windows